src/expansion.rs

```diff
@@ -1,25 +1,32 @@
 use super::peg::Job;
-use super::Variables;
+use super::Shell;
 
-pub fn expand_variables(jobs: &mut [Job], variables: &Variables) {
-    for mut job in &mut jobs[..] {
-        job.command = expand_string(&job.command, variables).to_string();
-        job.args = job.args
-                      .iter()
-                      .map(|original: &String| expand_string(&original, variables).to_string())
-                      .collect();
-    }
+pub trait Expand {
+    fn expand_variables(&self, jobs: &mut [Job]);
+    fn expand_string<'a>(&'a self, original: &'a str) -> &'a str;
```

Comment:
Does this have the unintended consequence of making `expand_string` public? It is meant to just be a helper method and should not be seen outside this module.